source/main.py

```
@@ -215,109 +223,92 @@ def save_to_file(path: str, lines: list[str]):
+
+def rename_vmess(link: str, ip: str, port: str, tag: str) -> str:
     try:
-        raw = link.split("ss://", 1)[1]
-        hash_tag = ""
-        if "#" in raw:
-            raw, hash_tag = raw.split("#", 1)
+        raw = link.split("://", 1)[1]
+        cfg = json.loads(b64_decode(raw))
+        cfg.update({"add": ip, "port": int(port), "ps": tag})
+        return f"vmess://{b64_encode(json.dumps(cfg))}#{quote(tag)}"
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)